master/buildbot/reporters/github.py

```diff
@@ -0,0 +1,138 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from buildbot.process.properties import Interpolate
+from buildbot.process.properties import Properties
+from buildbot.reporters import http
+
+from buildbot.process.results import CANCELLED
+from buildbot.process.results import EXCEPTION
+from buildbot.process.results import FAILURE
+from buildbot.process.results import RETRY
+from buildbot.process.results import SKIPPED
+from buildbot.process.results import SUCCESS
+from buildbot.process.results import WARNINGS
+from twisted.internet import defer
+from twisted.python import log
+
+HOSTED_BASE_URL = 'https://api.github.com'
+
+
+class GithubStatusPush(http.HttpStatusPushBase):
+    name = "GithubStatusPush"
+    neededDetails = dict(wantProperties=True)
+
+    @defer.inlineCallbacks
+    def reconfigService(self, token,
+                        startDescription=None, endDescription=None,
+                        context=None, baseURL=None, verbose=False, **kwargs):
+        yield http.HttpStatusPushBase.reconfigService(self, **kwargs)
+
+        self.context = context or Interpolate('buildbot/%(prop:buildername)s')
```

Comment:
should the `buildbot` part be hardcoded here?  what about making it a parameter of the service?